framework/src/utils/PetscSupport.C

```diff
@@ -385,7 +385,6 @@ petscNonlinearConverged(SNES snes,
                                         atol,
                                         nfuncs,
                                         maxf,
-                                        force_iteration,
```

Comment:
Some sudo code that should be put at around line 355

#if !PETSC_VERSION_LESS_THAN(3, 8, 4)
  ierr = SNESGetForceIteration(snes, &force_iteration);
  CHKERRABORT(problem.comm().get(), ierr);
#endif

if (force_iteration && !(problem.getNL_forced_its()))
  problem.setNL_forced_its(1)
  
if (!force_iteration && (problem.getNL_forced_its()) )
 {
  #if !PETSC_VERSION_LESS_THAN(3, 8, 4)
  ierr = SNESSetForceIteration(snes, PETSC_TRUE);
  CHKERRABORT(problem.comm().get(), ierr);
#endif
 }
  
  
  These is all you need

Reply:
yes this work